src/coreclr/src/jit/assertionprop.cpp

```diff
@@ -2624,13 +2624,6 @@ GenTree* Compiler::optConstantAssertionProp(AssertionDsc*        curAssertion,
 
         case O2K_CONST_LONG:
 
-#ifdef _TARGET_64BIT_
-            // Don't propagate handles if we need to report relocs.
-            if (opts.compReloc && ((curAssertion->op2.u1.iconFlags & GTF_ICON_HDL_MASK) != 0))
-            {
-                return nullptr;
-            }
-#endif
```

Comment:
You could add an assert here to insure that no change introduces a TYP_LONG witha reloc:

`assert((curAssertion->op2.u1.iconFlags & GTF_ICON_HDL_MASK) == 0);`

Reply:
How about adding this kind of check to `optDebugCheckAssertion`?

Reply:
Might make sense to merge this to unblock crossgen2 and add the assert in another PR.

Reply:
I am fine with that